src/systems/systems/src/electrical/emergency_generator.rs

```diff
@@ -70,6 +54,17 @@ impl EmergencyGenerator {
     fn should_provide_output(&self) -> bool {
         self.supplying
     }
+
+    fn update_generated_power(&mut self, gcu: &impl GeneratorControlUnitInterface) {
+        if gcu.hydraulic_motor_speed().get::<revolution_per_minute>()
+            > Self::MIN_RPM_TO_SUPPLY_POWER
+        {
+            self.generated_power =
+                Power::new::<watt>(Self::DEFAULT_POWER_DEMAND_WATT).min(gcu.max_allowed_power());
```

Comment:
**suggestion**: Instead of using a default power demand, consider using the actual power consumption (it will be one frame behind):

```rust
impl SimulationElement for EmergencyGenerator {
    fn process_power_consumption_report<T: PowerConsumptionReport>(
        &mut self,
        _: &UpdateContext,
        report: &T,
    ) {
        self.demand = report.total_consumption_of(PotentialOrigin::EmergencyGenerator);
    }
}
```

Reply:
Not sure how it's handled in electrical right now but getting what is consumed is not guaranteed though depending on hydraulic state.
But will definitly add some life on the generator once we plug real consumption in.

Reply:
Currently ELEC assumes whatever it consumes is supplied. In other words, if the GEN can supply less than ELEC consumes, everything will still "work". Not quite as it should, but works for now 😀.